crates/revm/src/handler/mainnet/pre_execution.rs

```diff
@@ -47,6 +48,56 @@ pub fn load_accounts<SPEC: Spec, EXT, DB: Database>(
         )?;
     }
 
+    // EIP-7702. Load bytecode to authorized accounts.
+    if SPEC::enabled(PRAGUE) {
+        if let Some(authorization_list) = context.evm.inner.env.tx.authorization_list.as_ref() {
+            let mut valid_auths = Vec::with_capacity(authorization_list.len());
+            for (authority, authorized) in authorization_list.iter() {
+                // TODO EIP-7702 do verification of authorizations.
+                // Recover authority and authorized addresses.
+
+                // warm authority account and check nonce.
+                let (authority_acc, _) = context
+                    .evm
+                    .inner
+                    .journaled_state
+                    .load_account(*authority, &mut context.evm.inner.db)?;
+
+                // TODO 2. Verify the chain id is either 0 or the chain's current ID.
+
+                // 3. Verify that the code of authority is empty.
+                // In case of multiple same authorities this step will skip loading of
+                // authorized account.
+                if authority_acc.info.code_hash() != KECCAK_EMPTY {
+                    continue;
+                }
+
+                // TODO 4. If nonce list item is length one, verify the nonce of authority is equal to nonce.
+
+                // warm code account and get the code.
+                // 6. Add the authority account to accessed_addresses
+                let (account, _) = context
+                    .evm
+                    .inner
+                    .journaled_state
+                    .load_code(*authorized, &mut context.evm.inner.db)?;
+                let code = account.info.code.clone();
+                let code_hash = account.info.code_hash;
+
+                // 5. Set the code of authority to code associated with address.
+                context.evm.inner.journaled_state.set_code_with_hash(
+                    *authority,
+                    code.unwrap_or_default(),
+                    code_hash,
+                );
+
+                valid_auths.push(*authority);
```

Comment:
per the spec:

>If any of the above steps fail, immediately stop processing that tuple and continue to the next tuple in the list. It will In the case of multiple tuples for the same authority, set the code specified by the address in the first occurrence.

so given the authorization here is valid, we should actually stop processing the next one. i.e. the first valid authorization is the one the tx is going to use (as far as i can parse)

Reply:
ah, maybe i got it wrong, i mixed up the address used for the code and the authority. so if i understand correctly now, you can have *multiple* valid authorities, but if there are two entries for the same authority, only the first one is used?

Reply:
> ah, maybe i got it wrong, i mixed up the address used for the code and the authority. so if i understand correctly now, you can have _multiple_ valid authorities, but if there are two entries for the same authority, only the first one is used?

True